chaosgcp/iam/controls/policy.py

```diff
@@ -0,0 +1,304 @@
+
+    Raises:
+        TypeError: If any of the arguments are not of the expected type.
+        ValueError: If `manage_type` is not "add" or "remove".
+        # (Add other potential exceptions based on your IAM implementation)
+
+    Example:
+        # Grant roles with a 15-minute expiration:
+        expiration_timestamp = datetime.datetime.now() + datetime.timedelta(minutes=15)
+        condition = {
+            "title": "TemporaryAccess",
+            "description": "Grants access for 15 minutes",
+            "expression": f"request.time < timestamp(\"{expiration_timestamp.strftime('%Y-%m-%dT%H:%M:%SZ')}\")",
+        }
+        manage_time_bound_temporary_iam_roles(
+            project_id="my-project",
+            members=["user1@example.com"],
+            roles=["roles/editor"],
+            roles_iam_condition=condition,
+            manage_type="add",
+        )
+    """
+    # Type checks for arguments
+    if not isinstance(project_id, str):
+        raise TypeError("project_id must be a string")
+    if not isinstance(members, list):
+        raise TypeError("members must be a list of strings")
+    if not isinstance(roles, list):
+        raise TypeError("roles must be a list of strings")
+    if not isinstance(roles_iam_condition, dict):
+        raise TypeError("roles_iam_condition must be a dictionary")
+    if not isinstance(manage_type, str):
+        raise TypeError("manage_type must be a string")
+    if not isinstance(iam_propogation_sleep_time_in_minutes, int):
+        raise TypeError(
+            "iam_propogation_sleep_time_in_minutes must be an integer"
+        )
+
+    # Initializes service.
+    crm_service = initialize_service()
+
+    if manage_type == "add":
+        logger.info("Adding Conditional Roles..")
+        logger.info(roles_iam_condition)
+        modify_policy_add_roles(
+            crm_service,
+            project_id,
+            roles,
+            members,
+            roles_iam_condition,
+            iam_propogation_sleep_time_in_minutes,
+        )
+
+    elif manage_type == "remove":
+        logger.info(roles_iam_condition)
+        logger.info("Removing Conditional Roles..")
+        modify_policy_remove_roles(
+            crm_service, project_id, roles, members, roles_iam_condition
+        )
+
+    else:
+        logger.warning("Invalid manage type: must be 'add' or 'remove'.")
+        raise ValueError("Invalid manage type: must be 'add' or 'remove'")
+
+
+def initialize_service() -> dict:
+    """Initializes a Cloud Resource Manager service."""
+
+    credentials, _ = google.auth.default(
+        scopes=["https://www.googleapis.com/auth/cloud-platform"]
+    )
+    crm_service = googleapiclient.discovery.build(
+        "cloudresourcemanager", "v1", credentials=credentials
+    )
+    return crm_service
+
+
+def get_policy(crm_service, project_id: str, version: int = 3) -> dict:
+    """Gets IAM policy for a project."""
+    logger.info("Getting IAM policy for a project.")
```

Comment:
We tend to keep the `info` level only for experiment related necessary logs. I don't mind that we use `info` but if this log is more "debug" oriented, then it should be `.debug(...)` instead.

Reply:
It is info related , because user should know about the policy changes and can act accordingly if something goes wrong .